Zend/zend_language_scanner.l

```diff
@@ -2569,6 +2663,7 @@ skip_escape_conversion:
 <ST_DOUBLE_QUOTES,ST_BACKQUOTE,ST_HEREDOC>"{$" {
 	yy_push_state(ST_IN_SCRIPTING);
 	yyless(1);
+	enter_nesting(yytext);
```

Comment:
This will refer to an opening `{` rather than `{$`. That's fine, but makes me wonder if we need the special-casing around `${`? Shouldn't we just report that as `{` as well?

Reply:
I think we still need the special-casing for `${`. The error messages will be more helpful. The case where `${` will be reported is the one on line 1878 (different from this one).

Reply:
This makes the error reporting inconsistent though. It means that `${` outside a string will report as an unclosed `{`, while a `${` inside a string will report and unclosed `${`. This seems somewhat odd to me. The fact that we treat `$` and `{` as separate tokens outside strings and as a single token inside strings shouldn't leak through in this way.

Reply:
Maybe my intuition is off, but when I see `"... ${<expression>} ..."`, intuitively I think of the `${` as matching the `}`. In my mind, those are the delimiters. But when I see something like `$foo->{$bar}`, I think of `{` as matching `}` and `$bar` as being the "contents" of the braces. To my eyes, `{$` doesn't appear to be a delimiter, but rather just `{`.

(Please note that above you refer to "`${` outside a string", which is not what we are dealing with here. It's `{$` outside a string.)

Anyways, please just tell me what you want. Would you like PHP to report `{$` as not matching a different (or missing) closing delimiter? If so, that's fine. I can do that.

Reply:
> (Please note that above you refer to "`${` outside a string", which is not what we are dealing with here. It's `{$` outside a string.)

To be clear, I'm talking about something like `${$foo}`. This is valid both inside and outside strings, and I think those should receive the same treatment, either by referring to `${` in both cases, or `{` in both cases. As writing `$ {$foo}` with additional whitespace is also legal (outside strings), I'd go with just reporting the `{` in both cases.

Reply:
OK. That makes sense now.

One hot, fresh little force-push coming right up!!!